src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -49,6 +53,9 @@ public FunctionMetadataManager(IOptions<ScriptJobHostOptions> scriptOptions, IFu
             _isHttpWorker = httpWorkerOptions?.Value?.Description != null;
             _environment = environment;
 
+            _languageOptions = languageOptions;
+            _onChangeSubscription = languageOptions.OnChange(_ => _servicesReset = true);
```

Comment:
Should updating this class level local variable be inside a lock or `Interlocked.Exchange`? I am wondering there could be a race condition if the `OncChange` and `GetFunctionMetdata` public method gets invoked at the same time which might cause any sort of unintentional behavior.

Reply:
Read and write of boolean is atomic, so no need to wrap in `Interlocked`.